### PR TITLE
systemd service: security hardening

### PR DESCRIPTION
**Pull request by stapelberg**
I used `systemd-analyze security authelia` and enabled a number of options that significantly reduce the attack surface. My instance of authelia still works with these options enabled, but comments welcome as to whether any of the settings might go a little too far…?

## Summary by CodeRabbit

- **Security Enhancements**
	- Enhanced security measures by restricting various capabilities, privileges, and access controls in service operations.

### CODE REVIEW

**Comment by james-d-elliott** [comment]
Thanks for contributing. I think this should be okay. That being said I'm no systemd expert. I'll see if anyone else has any thoughts.

**Comment by james-d-elliott** [comment]
@coderabbitai please review this PR.

**Comment by nightah** [comment]
There's nothing harmful in the presented changes, I would suggest removing all the newlines and commentary for each option though.

Remember this file is only provided as a baseline because not all users would necessarily deploy Authelia via packaged mechanisms (some will just use the binary and the provided systemd unit).

File: authelia.service [hunk 1]
```diff
@@ -7,5 +7,39 @@ Environment=AUTHELIA_SERVER_DISABLE_HEALTHCHECK=true
...
+ProtectKernelModules=yes
+
+# Make kernel variables (e.g. /proc/sys) read-only:
+ProtectKernelTunables=yes
+
+# Filter dangerous system calls. The following is listed as safe basic choice
+# in systemd.exec(5):
+SystemCallArchitectures=native
+SystemCallFilter=@system-service
+SystemCallErrorNumber=EPERM
```
Comment (james-d-elliott) [inline authelia.service:42]:
```suggestion
CapabilityBoundingSet=
NoNewPrivileges=yes
RestrictNamespaces=yes
ProtectHome=true
PrivateDevices=yes
PrivateUsers=yes
ProtectControlGroups=yes
ProtectKernelModules=yes
ProtectKernelTunables=yes
SystemCallArchitectures=native
SystemCallFilter=@system-service
SystemCallErrorNumber=EPERM
```

**Review by james-d-elliott — CHANGES_REQUESTED** [review]

**Review by james-d-elliott — APPROVED** [review]

**Pull request merged**